src/StreamJsonRpc/MessageHandlerBase.cs

```diff
@@ -35,6 +35,26 @@ public abstract class MessageHandlerBase : IJsonRpcMessageHandler, IDisposableOb
         /// </summary>
         private readonly AsyncSemaphore sendingSemaphore = new AsyncSemaphore(1);
 
+        /// <summary>
+        /// The sync object to lock when inspecting and mutating the <see cref="state"/> field.
+        /// </summary>
+        private readonly object syncObject = new object();
+
+        /// <summary>
+        /// A signal that the last read operation has completed.
+        /// </summary>
+        private readonly AsyncManualResetEvent readingCompleted = new AsyncManualResetEvent();
+
+        /// <summary>
+        /// A signal that the last write operation has completed.
+        /// </summary>
+        private readonly AsyncManualResetEvent writingCompleted = new AsyncManualResetEvent();
+
+        /// <summary>
+        /// A value indicating whether the <see cref="ReadAsync(CancellationToken)"/> method is in progress.
```

Comment:
The comment seems a little off, as this state indicates whether a ReadAsync and/or a WriteAsync method is in progress

Reply:
Fixed